google/generativeai/types/content_types.py

```diff
@@ -650,12 +671,56 @@ def _encode_fd(fd: FunctionDeclaration | protos.FunctionDeclaration) -> protos.F
     return fd.to_proto()
 
 
+GoogleSearchRetrievalType = Union[protos.GoogleSearchRetrieval, dict[str, float]]
+
+
+def _make_google_search_retrieval(gsr: GoogleSearchRetrievalType):
+    if isinstance(gsr, protos.GoogleSearchRetrieval):
+        return gsr
+    elif isinstance(gsr, Iterable) and not isinstance(gsr, Mapping):
```

Comment:
I'm skeptical of this branch.

1. I don't see `Sequence` in `GoogleSearchRetrievalType`.
2. How would a list get here?
3. How do we know it's not a list of something else?